docs/notebooks/multiobjective_ehvi.pct.py

```diff
@@ -0,0 +1,176 @@
+
+models = {OBJECTIVE: ModelStack(*objective_models)}
+
+# ## Define the acquisition function
+# Here we utilize the [EHVI](https://link.springer.com/article/10.1007/s10898-019-00798-7): `ExpectedHypervolumeImprovement` acquisition function:
+
+from trieste.acquisition.rule import EfficientGlobalOptimization
+
+ehvi = ExpectedHypervolumeImprovement()
+rule: EfficientGlobalOptimization[Box] = EfficientGlobalOptimization(builder=ehvi.using(OBJECTIVE))
+
+# ## Run the optimization loop
+#
+# We can now run the optimization loop
+
+num_steps = 20
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+result = bo.optimize(num_steps, initial_data, models, acquisition_rule=rule)
+
+# To conclude, we visualize the queried data across the design space.
+
+# +
+datasets = result.try_get_final_datasets()
+data_query_points = datasets[OBJECTIVE].query_points.numpy()
+data_observations = datasets[OBJECTIVE].observations.numpy()
+
+_, ax = plot_function_2d(
+    vlmop2, mins, maxs, grid_density=100, contour=True, title=["Obj 1", "Obj 2"]
+)
+plot_bo_points(data_query_points, ax=ax[0, 0], num_init=num_initial_points)
+plot_bo_points(data_query_points, ax=ax[0, 1], num_init=num_initial_points)
+plt.show()
+# -
+
+# ... and visulize in the objective space, purple dots denotes the nondominated points.
+
+plot_mobo_points_in_obj_space(data_observations, num_init=num_initial_points)
+plt.show()
+
+# We can also visualize how the performance metric get envolved with respect to the number of BO iterations.  
+# First, we need to define a performance metric. Many metrics have been considered for multi-objective function. Here, we use the log hypervolume difference.
+
+# The log hypervolume difference is defined as the difference between the hypervolume of the true Pareto front and the hypervolume of the approximate Pareto front based on the observed data.
+# $$
+# log_{10}\ \text{HV}_{diff} = log_{10}(\text{HV}_{\text{true}} - \text{HV}_{\text{bo obtained}})
+# $$
+#
+
+# First we need to calculate the $\text{HV}_{\text{ideal}}$, this is approximated by calculating the hypervolume based on ideal pareto front:
+
+ideal_pf = (VLMOP2().gen_pareto_optimal_points(100)
+)  # gen 100 pf points
+ref_point = get_reference_point(data_observations)
+idea_hv = Pareto(tf.cast(ideal_pf, dtype=data_observations.dtype)).hypervolume_indicator(ref_point)
+
+
+# Then define the metric function:
+
+
+def log_hv(observations):
+    obs_hv = Pareto(observations).hypervolume_indicator(ref_point)
+    return math.log(idea_hv - obs_hv)
+
+
+fig, ax = plot_mobo_history(data_observations, log_hv, num_init=num_initial_points)
+ax.set_xlabel("Iterations")
+ax.set_ylabel("log HV difference")
+plt.show()
+
+mask_fail = np.linspace(1, 20, 20).astype(int)
+fig, ax = plot_mobo_history(data_observations, log_hv, num_init=num_initial_points, mask_fail=mask_fail)
```

Comment:
Whats the difference between these two plots?

Reply:
Removed the latter one as originally used for `mask_fail` test only.